### PR TITLE
feat: include metadata in iamv1.Caller

### PR DESCRIPTION
**Pull request by odsod**
For permission tester implementations that want to do caching.

To facilitate this, iammember.ResolveResult has been consolidated into
the iamv1.Caller proto message.

BREAKING CHANGE: Updates to the iammember API, moving the Resolver
                 interface to the new iamcaller package.

### CODE REVIEW

File: iamcaller/caller.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+				continue MemberLoop
+			}
+		}
+		caller.Members = append(caller.Members, member)
+	}
+	sort.Slice(caller.Members, func(i, j int) bool {
+		return caller.Members[i] < caller.Members[j]
+	})
+	if caller.Metadata == nil {
+		caller.Metadata = map[string]*iamv1.Caller_Metadata{}
```
Comment (ericwenn) [inline iamcaller/caller.go:24]: nit:
```suggestion
		caller.Metadata = make(map[string]*iamv1.Caller_Metadata)
```

File: iamcaller/chain_test.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"google.golang.org/protobuf/testing/protocmp"
+
+	iamv1 "go.einride.tech/iam/proto/gen/einride/iam/v1"
+
+	"gotest.tools/v3/assert"
```
Comment (ericwenn) [inline iamcaller/chain_test.go:12]: imports

File: iamcaller/validate.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+			return err
+		}
+		// Validate that the member is present in the metadata.
+		for _, metadata := range caller.Metadata {
+			for _, metadataMember := range metadata.Members {
+				if member == metadataMember {
+					continue ValidateMembersLoop
+				}
+			}
+		}
```
Comment (ericwenn) [inline iamcaller/validate.go:35]: IMO move this to another function and remove the `ValidateMembersLoop` label.
Maybe `func hasMember(metadata *iamv1.Metadata, member string) bool`

File: iamcaller/validate.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+			if err := iammember.Validate(metadataMember); err != nil {
+				return err
+			}
+			// Validate that the metadata member is present in the top-level members.
+			for _, callerMember := range caller.Members {
+				if metadataMember == callerMember {
+					continue ValidateMetadataLoop
+				}
+			}
+			return fmt.Errorf("member '%s' from metadata '%s' not in caller members", metadataMember, key)
```
Comment (ericwenn) [inline iamcaller/validate.go:55]: Same here as above. Labels makes it harder to understand IMO.

**Review by ericwenn — APPROVED** [review]

File: iamcaller/validate.go (hunk 3 above)
Comment (odsod) [reply iamcaller/validate.go:35]: Good point! Fixing in a follow-up!

**Pull request merged**